src/sinks/http.rs

```diff
@@ -215,7 +216,11 @@ impl HttpSink for HttpSinkConfig {
             byte_size: body.len(),
         });
 
-        Some(body)
+        let (_, metadata) = event.into_parts();
+        Some(EncodedEvent {
+            item: body,
+            metadata: Some(metadata),
+        })
```

Comment:
This is the actual change that will have to be done to the other HTTP-based sinks. This should be trivial for at least most of the logs sinks, but impossible without a clone for metrics sinks.